src/modules/75-cf/components/GitSyncActions/GitSyncActions.tsx

```diff
@@ -18,10 +19,17 @@ export interface GitSyncActionsProps {
 }
 
 const GitSyncActions = ({ isLoading }: GitSyncActionsProps): ReactElement => {
-  const { gitRepoDetails, isGitSyncActionsEnabled } = useFFGitSyncContext()
+  const { gitRepoDetails, isGitSyncEnabled, gitSyncLoading } = useFFGitSyncContext()
   const [isSettingsOpen, setIsSettingsOpen] = useState(false)
 
-  return isGitSyncActionsEnabled ? (
+  if (gitSyncLoading) {
+    return (
+      <Container flex={{ align: 'center-center' }} width="100px">
+        <Spinner size={24} data-testid="git-sync-spinner" />
+      </Container>
+    )
+  }
+  return isGitSyncEnabled ? (
```

Comment:
Super minor, but a line break here would help make this a bit more readable.
```suggestion

  return isGitSyncEnabled ? (
```